R/checkResult.R

```diff
@@ -0,0 +1,81 @@
+
+
+
+checkResult <- R6::R6Class(
```

Comment:
I'm inclined to leave it as an R6 object, I do agree with your point but using the object allows us validate the data input and do updates like setting nulls instead of missings.  

Following your comments I tried going to a named list but it breaks all the unit tests for the individual functions which rely on the data parsing properties of the class.  

I don' think its doing any harm keeping it....

Reply:
I mean I guess you could replace it with a function that does the same validation work but then why not just leave it as an R6 object as its just achieving the same thing ? In theory the R6 object provides more guarantees as you can apply an assertion that the returned object is R6 and has therefore had the validation steps applied .

Reply:
Yeah my thinking was to just make a function which does the validation. If we really wanted to check if it's been validated we can just give it a class to indicate that (checked_object). The main reason to avoid R6 is just to avoid complexity where it's not needed, for other people reading our code in future. There's also probably a small performance hit from making this, although I can't imagine its significant...

Reply:
I think my logic for this is that as we are using R6 I would rather use it consistently through out  so we aren't mixing and matching OO systems.  Here as an example we have a function thats highly related to a particular data source so to me it makes sense to have it as an object. Can change it if you feel super strong about it but to me the current implementation seems reasonable.